cloud/amazon/ec2_eni.py

```diff
@@ -0,0 +1,406 @@
+#!/usr/bin/python
+#
+# This is a free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This Ansible library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: ec2_eni
+short_description: Create and optionally attach an Elastic Network Interface (ENI) to an instance
+description:
+    - Create and optionally attach an Elastic Network Interface (ENI) to an instance. If an ENI ID is provided, an attempt is made to update the existing ENI. By passing 'None' as the instance_id, an ENI can be detached from an instance.
+version_added: "1.9"
+author: Rob White, wimnat [at] gmail.com, @wimnat
+options:
+  instance_id:
+    description:
+      - Instance ID that you wish to attach ENI to 
+    required: false
+    default: null 
+    aliases: []
+  private_ip_address:
+    description:
+      - Private IP address.
+    required: false
+    default: null
+    aliases: []
+  subnet_id:
+    description:
+      - Subnet in which to create the ENI
+    required: true
+    default: null
```

Comment:
@bcoca subnet_id is required for state: present but not for state: absent.  What's the best way to document?